go-controller/pkg/node/gateway_localnet.go

```diff
@@ -116,32 +122,107 @@ func generateGatewayNATRules(ifname string, ip net.IP) []iptRule {
 	return rules
 }
 
-func localnetGatewayNAT(ipt util.IPTablesHelper, ifname string, ip net.IP) error {
-	rules := generateGatewayNATRules(ifname, ip)
-	return addIptRules(ipt, rules)
+func localnetGatewayNAT(localNetdata []*localnetData, ifname string) error {
```

Comment:
(should probably either move the definition of `localnetData` to before here, or else move this function to after the definition of `localnetData`)